### PR TITLE
Ci

### PR DESCRIPTION
**Pull request by 3b**
was going to make separate PRs, but got messy once i started doing CI stuff. let me know if you want only some of it

* implement bits-short-float and short-bits-float for platforms where short-float = single-float
* fixes some problems with bits-*-float and *-float-bits with negative floats on various platforms
* optimize bits-*-float a bit on sbcl
* add tests and run them in CI

### CODE REVIEW

**Comment by Shinmera** [comment]
Looks good. The only issue I have is with the random tester portion, since it won't actually report the failing test if there is one and just say the loop fails. That's a problem with Parachute though -- don't want to spam the output with thousands of random forms, either. I suppose for now this is fine, will have to go back once Parachute's support for stuff like this is better.

File: float-features.lisp [hunk 1]
```diff
@@ -261,26 +263,61 @@
...
 (defun short-float-bits (float)
-  #-mezzano
-  (declare (ignore float))
+  (declare (ignorable float))
   #+mezzano
   (mezzano.extensions:short-float-to-ieee-binary16 float)
-  #-mezzano
-  (error "Implementation not supported."))
+  #+(or ecl sbcl cmucl allegro ccl
+        (and 64-bit lispworks))
```
Comment (Shinmera) [inline float-features.lisp:270]: Any reason 32-bit lispworks is excluded here?

Comment (3b) [reply float-features.lisp:270]: see comments in short-float-bits.
32bit lw has separate short-float and single-float types, and i was too lazy to make it generic for that and clisp :)

**Comment by 3b** [comment]
the loops aren't random, can do exhaustive tests on 16bit floats easily
64k tests (twice) would be a bit verbose in the logs though

File: float-features.lisp (hunk 1 above)
Comment (Shinmera) [reply float-features.lisp:270]: Ah. weird that they would have that only on 32 bit then!

**Comment by Shinmera** [comment]
Right, my initial point stands -- Parachute needs some way of compacting tests or hiding them. Or, even better, more general support for procedural tests.

**Comment by 3b** [comment]
64 bit has space for tagged single floats, so no reason to have a smaller
short float type

**Comment by Shinmera** [comment]
There is, though! 16 bit floats are used extensively these days in stuff like NNs.

**Comment by 3b** [comment]
No reason to keep the "single float with some bits chopped off for tagging"
short-float.

16 bit short floats would be nice, though smaller than suggested size in CL
spec.

**Pull request merged**